### PR TITLE
drivers: wifi: Optimize the memory consumed by logs

### PR DESCRIPTION
**Pull request by rado17**
Limit the information displayed in logging (function, line etc.). Disable printing of debug message by default.

### CODE REVIEW

**Review by VivekUppunda — APPROVED** [review]

File: drivers/wifi/nrf700x/Kconfig [hunk 1]
```diff
@@ -70,6 +70,10 @@ config NRF700X_REG_DOMAIN
 config NET_MGMT_EVENT_STACK_SIZE
 	default 2048 if !WPA_SUPP
 
+config NRF700X_LOG_VERBOSE
+	bool "Maintains the verbosity of information in logs"
+	default n
```
Comment (krish2718) [inline drivers/wifi/nrf700x/Kconfig:75]: not needed, default is `default n`

File: modules/hostap/Kconfig [hunk 2]
```diff
@@ -170,4 +170,5 @@ config LOG_BUFFER_SIZE
 
 config WPA_SUPP_NO_DEBUG
 	bool "Disable printing of debug messages, saves code size significantly"
+	default y
```
Comment (krish2718) [inline modules/hostap/Kconfig:173]: This should be a separate commit.

Comment (krish2718) [reply modules/hostap/Kconfig:173]: And I don't think default should be no debugs, this hinders customer support.

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: modules/hostap/Kconfig (hunk 2 above)
Comment (rado17) [reply modules/hostap/Kconfig:173]: In that case, shall we enable this specifically in all matter related samples for now?

Comment (krish2718) [reply modules/hostap/Kconfig:173]: Yes, each sample can decide individually.

Comment (rado17) [reply modules/hostap/Kconfig:173]: Done

File: drivers/wifi/nrf700x/Kconfig (hunk 1 above)
Comment (rado17) [reply drivers/wifi/nrf700x/Kconfig]: Removed.

**Review by sachinthegreen — APPROVED** [review]

**Comment by rado17** [comment]
> For Matter we have `WPA_SUPP_NO_DEBUG` already set by default:
> 
> https://github.com/nrfconnect/sdk-connectedhomeip/blob/master/config/nrfconnect/chip-module/Kconfig.defaults#L250

Will drop this commit.

**Review by LuDuda — APPROVED** [review]

**Comment by LuDuda** [comment]
Thank you for this improvement! :+1:

As discussed F2F - introduction of `CONFIG_NRF700X_LOG_VERBOSE` is a good improvement for debug variants. Still however, we don't have control over log level in the Wi-Fi OSAL layer. E.g. even if LOGS are completely off (release build), log strings will occupy the flash memory. It would be perfect to have Zephyr based logging approach there.

**Review by LuDuda — APPROVED** [review]
Awesome!

File: modules/hostap/Kconfig (hunk 2 above)
Comment (rado17) [reply modules/hostap/Kconfig]: Right. This was inadvertantly included. Removed now.

**Review by krish2718 — APPROVED** [review]
